saleor/graphql/order/utils.py

```diff
@@ -0,0 +1,41 @@
+from ..core.types.common import Error
+
+
+def check_for_draft_order_errors(order, errors):
+    """Return a list of errors associated with the order.
+
+    Checks, if given order has a proper customer data, shipping
+    address and method set up and return list of errors if not.
+    """
+    if order.get_total_quantity() == 0:
+        errors.append(
+            Error(
+                field='lines',
+                message='Could not create order without any products.'))
+    if order.is_shipping_required():
+        method = order.shipping_method
+        shipping_address = order.shipping_address
+        shipping_not_valid = (
+            method and shipping_address and
+            shipping_address.country.code not in method.shipping_zone.countries)  # noqa
+        if shipping_not_valid:
+            errors.append(
+                Error(
+                    field='shipping',
+                    message='Shipping method is not valid for chosen shipping '
+                            'address'))
+    if not order.user and not order.user_email:
+        errors.append(
+            Error(
+                field=None,
+                message='Both user and user_email fields are null'))
+    return errors
+
+
+def can_finalize_order(order):
+    """Return `True` if an order draft can be complete.
```

Comment:
Have you checked this docstring with `pydocstyle`? I think it's not in correct format.

Reply:
It seems that `check_for_draft_order_errors` function tells us whether a draft order can be finalized. If yes I'd suggest renaming `check_for_draft_order_errors` to `can_finalize_draft_order` and dropping `can_finalize_order` at all.